### ISSUE TITLE
challenge participant lists / search should use Username not Display Name

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#11456]
Ceran, Star Keeper / @Ceran (aabe567b-aa55-42c1-b612-b7699a7ba9b7) points out that that challenge winner selection might be easier if it used Usernames instead of Display Names. I've copied their post in full below. I'll leave this as suggestion-discussion for a few days and then if there's no objections, I'll mark it as help wanted to get changes made to <strike>the CSV export files and to</strike> the winner selection UI elements in the website.

--------------------------------------------

"When awarding challenges via the website, the drop-down menu of participants to chose the winner from shows their display names. I think it might be worth considering changing this to show the user names instead, because the display names showing causes issues with awarding:

1. If I don't get around to award the winner immediately after downloading the challenge datas and the person who won changes their display name in the meantime, I can't find it in the list anymore and have to go hunt down the person's new name, which eats up needless time.
1. If someone has a display name that consists out of non-latin letters, some special looking latin script or emojis, most of the time it can not be displayed in the list and these parts are cut out, turned into a garble or other characters or replaced by empty boxes. That makes it hard to recognize the person in the list. Often I can only find them because "since they have an unusual display name, the array of empty boxes must be them".
1. The downloadable data often also has this problem with displaying non-standard latin characters. If a challenge has a lot of participants, and the person who won does not appear right at the top of the drop-down menu for awarding, I have to type in the person's name in the box to make it show. If it consists of mentioned letters however, I am unable to type the name, and since the spreadsheet has the same problem, I can not copypaste their name from their either, nor from their user profile, since only the messed up version shows in the drop-down menu anyway. In this case I have to go over the challenge data tool to be able to award a winner.

In summary, I think all of these issues could be easily avoided if the participants list to pick the winner from was changed to show the user name instead of the display name."

**Comment by adamfitzgibbon** [comment]
Seems reasonable, I'll take a look at this one!

**Comment by adamfitzgibbon** [comment]
Note on this, the "Export Challenge" functionality already includes username. I think the reporter meant that they were unable to copy and paste the non-latin characters from the csv since they were also messed up. Perhaps this issue can be fixed if we used a character-set that included these characters? For now I'll stick to updating the members drop-down and search functionality to use username.

**Comment by Alys** [comment]
@adamfitzgibbon Thanks! I've marked this in progress for you.

Sorry about the confusion with the export file. It's fine to change just the website features as you've described. I'll edit this issue's title to remove the mention of the CSV file. Let's not worry about looking into changing the character set for this since using Usernames will bypass that issue.

**Comment by adamfitzgibbon** [comment]
No problem, sounds good! That makes it a pretty minor change so here's a small PR to address it.
